src/sys/ioctl/platform/bsd.rs

```diff
@@ -0,0 +1,41 @@
+mod consts {
+    pub const VOID: u32 = 0x20000000;
+    pub const OUT: u32 = 0x40000000;
+    pub const IN: u32 = 0x80000000;
+    pub const INOUT: u32 = (IN|OUT);
+    pub const IOCPARM_MASK: u32 = 0x1fff;
+}
+
+pub use self::consts::*;
+
+#[macro_export]
+macro_rules! ioc {
+    ($inout:expr, $group:expr, $num:expr, $len:expr) => (
+        $inout | (($len as u32 & $crate::sys::ioctl::IOCPARM_MASK as u32) << 16) | (($group as u32) << 8) | ($num as u32)
```

Comment:
The second `as u32` is superfluous, since `IOCPARAM_MASK` is already defined as `u32`.